vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java

```diff
@@ -144,14 +145,23 @@ public static ConfigurationResolver getInstance() throws IOException
         deprecatedPropertiesHandler.removeDeprecated(properties);
         resolveSpelExpressions(properties, false);
 
-        try (VaultStoredPropertiesProcessor vaultProcessor = new VaultStoredPropertiesProcessor(properties))
-        {
-            PropertiesProcessor propertiesProcessor = new DelegatingPropertiesProcessor(List.of(
-                    new EncryptedPropertiesProcessor(properties), vaultProcessor
-            ));
-            new SystemPropertiesInitializer(propertiesProcessor).setSystemProperties(properties);
-            properties = propertiesProcessor.processProperties(properties);
-        }
+        ServiceLoader<PropertiesProcessor> loader = ServiceLoader.load(PropertiesProcessor.class);
+        PropertiesProcessor propertiesProcessor = new DelegatingPropertiesProcessor(loader);
+        properties = propertiesProcessor.processProperties(properties);
+        new SystemPropertiesInitializer(propertiesProcessor).setSystemProperties(properties);
+        loader.forEach(processor -> {
+            if (processor instanceof VaultStoredPropertiesProcessor)
```

Comment:
`ConfigurationResolver` should know nothing about particular implementations of `PropertiesProcessor`, it should be designed in this way:
- `PropertiesProcessor` should inherit `Closeable` interface
- `PropertiesProcessor` should have default empty implementation of `close()`
- `DelegatingPropertiesProcessor` in its `close` method should invoke `close` for each delegate
- `DelegatingPropertiesProcessor` instance should be used with `try-with-resources`